### PR TITLE
WIP/Draft: KGPE-D16 re-support

### PR DESCRIPTION
**Pull request by Tonux599**
Expect 6ish months to complete due to other commitments.

Primary work/testing will be on the `kgpe-d16_workstation` variant but other can be re-enabled if others want to test them.

Todo List:
- [ ] Order more DIP8 chips and setup work lab.
- [ ] Build without error against latest Dasharo.
- [ ] Use more up-to-date Linux kernel (as maybe Aspeed BMC video improvements that are relevant? https://github.com/linuxboot/heads/commit/12c7dfdadc02f9fe36c058c175a9c7a726b43932).
- [ ] Update coreboot/dasharo/linux configs accordingly.
- [ ] Investigate and fix the flashrom problem. https://github.com/linuxboot/heads/issues/1395#issuecomment-1763182318 https://github.com/linuxboot/heads/issues/1395#issuecomment-1533590357
- [ ] Either support TPM1.2 and TPM2.0 on the same board config. Else, setup multiple variants for TPM1.2 and TPM2.0.
- [ ] Investigate potentially maintaining a legacy coreboot 4.11 variant with TPM1.2 due to performance issues with latest Dasharo.
- [ ] Verify on all variants that boot is measured and PCR's are filled.
- [ ] (*BIG MAYBE*) investigate if simpledrm/simplefb/efifb can replace the need to include AMD/Nvidia kernel modules and binary blobs for recent GPUs. https://github.com/linuxboot/heads/issues/1395#issuecomment-1718268849
- [ ] Ensure all variants build on CI.
- [ ] **PRE-MERGE: re-enable all other boards in CI** (disabled while testing) - revert 988aaf1643d90fb3ff8a068abd264c48a20d51b4

### CODE REVIEW

**Comment by tlaurion** [comment]
@Tonux599 on tpm support, boards need to be either TPM1 or tpm2 since toolstack is different.

You can look at qemu boards for examples

**Comment by tlaurion** [comment]
@Tonux599

> Investigate potentially maintaining a legacy coreboot 4.11 variant with TPM1.2 due to performance issues with latest Dasharo.

ELI5? SoA?

**Comment by tlaurion** [comment]
@Tonux599 ref of vikings HCL over 4.11

**Comment by Tonux599** [comment]
> @Tonux599
> 
> > Investigate potentially maintaining a legacy coreboot 4.11 variant with TPM1.2 due to performance issues with latest Dasharo.
> 
> ELI5? SoA?

Define SoA?

But I'm not too sure to be honest, I've heard that Dasharo is slower at runtime. Once my testing suit is setup I'll run some benchmarks and try and reproduce.

**Comment by tlaurion** [comment]
>  Add vikings HCL link as comment to board configs.

Checked per #1759 being superseded as #1761 I just merged.

**Comment by tlaurion** [comment]
> > @Tonux599
> > > Investigate potentially maintaining a legacy coreboot 4.11 variant with TPM1.2 due to performance issues with latest Dasharo.
> > 
> > 
> > ELI5? SoA?
> 
> Define SoA?
> 
> But I'm not too sure to be honest, I've heard that Dasharo is slower at runtime. Once my testing suit is setup I'll run some benchmarks and try and reproduce.

SoA: State Of the Art.
@Tonux599 also, https://github.com/linuxboot/heads/pull/1760 brings raminit changes merged into Dasharo back to coreboot 4.11: it needs testing and report on improvement/regression.

**Comment by tlaurion** [comment]
@Tonux599 ping

**Comment by tlaurion** [comment]
See #1910 @Tonux599
